python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1294,6 +1377,9 @@ def get_tag_filter_query(tag_key, tag_value):
                 )
             )
 
+        if filter_event_id is not None:
```

Comment:
Can we move these lines within the if-else block above? so
```
if not filter_tags:
            tags_query = (
               ...
            )
            if filter_event_id is not None:
                 tags_query = tags_query.where(AssetEventTagsTable.c.event_id == filter_event_id)
```
and 
```
def get_tag_filter_query(...)
     filter_query = ....
      if filter_event_id is not None:
             filter_query = filter_query.where(AssetEventTagsTable.c.event_id == filter_event_id)
```
Mostly just because the intersect query may be more expensive and I think it would be good to filter out undesired event IDs before the intersect

Reply:
I think the query optimizer inside of postgres will handle that sort of optimization for us 🤞